### PR TITLE
feat(sdk): add CreateFromListing method for databases

### LINKED ISSUE #4047: [Feature]: create database from listing
### Company Name

_No response_

### Use Cases or Problem Statement

My team is migrating from database replication to organizational listings for cross-region data sharing.
However I don't see terraform provider support for operations like `CREATE DATABASE <name> FROM LISTING '<listing_global_name>'`.

### Category

category:resource

### Object type(s)

resource:database

### Proposal

_No response_

### How much impact is this issue causing?

Medium

### Additional Information

_No response_

### Would you like to implement a fix?

- [ ] Yeah, I'll take it 😎

### PR DESCRIPTION
**Pull request by ghribikhawla-prog**
## Summary
Add `CreateDatabaseFromListingOptions` struct and `CreateFromListing` SDK method to support `CREATE DATABASE <name> FROM LISTING '<listing_global_name>'` syntax.

Closes #4047

## Changes (SDK only, Step 1 per [CONTRIBUTING.md](https://github.com/snowflakedb/terraform-provider-snowflake/blob/dev/CONTRIBUTING.md#adding-support-for-a-new-snowflake-object))
- **pkg/sdk/databases.go**: Add `CreateDatabaseFromListingOptions` struct, validation, and `CreateFromListing` method
- **pkg/sdk/databases_test.go**: Add unit tests (nil options, invalid name, empty listing, SQL generation)
- **pkg/sdk/testint/databases_integration_test.go**: Add integration test (requires `TEST_SF_TF_LISTING_GLOBAL_NAME` env var)

## Testing
- Unit tests: `go test ./pkg/sdk/ -run TestDatabasesCreateFromListing -v` — 4/4 PASS
- Integration test: gated behind `TEST_SF_TF_LISTING_GLOBAL_NAME` env var
- `go build ./...` passes cleanly

## Note
This PR contains only the SDK changes (Step 1). The Terraform resource and related configuration will follow in subsequent PRs as outlined in the contributing guide.

### CODE REVIEW

**Comment by ghribikhawla-prog** [comment]
@sfc-gh-jcieslak Thanks for the review on #4741! I've addressed all your feedback in this new PR:
- ✅ Target branch is now `dev`
- ✅ SDK-only scope (no resource, no provider, no preview features)
- ✅ Unit tests + integration test included

Would you mind taking a look? Thanks!

File: pkg/sdk/testint/databases_integration_test.go [hunk 1]
```diff
@@ -337,6 +338,32 @@ func TestInt_DatabasesCreateSecondary(t *testing.T) {
...
 
+func TestInt_DatabasesCreateFromListing(t *testing.T) {
+	client := testClient(t)
+	ctx := testContext(t)
+
+	// This test requires a listing to be available on the consumer account.
+	// The listing global name must be set via the TEST_SF_TF_LISTING_GLOBAL_NAME environment variable.
+	// To set up a listing for testing, create a private listing on a provider account
+	// and share it with the consumer account used for integration tests.
+	listingGlobalName := os.Getenv("TEST_SF_TF_LISTING_GLOBAL_NAME")
```
Comment (sfc-gh-jcieslak) [inline pkg/sdk/testint/databases_integration_test.go]: We have tests that create listings (`listings_integration_test.go`). You can try and leverage testClientHelper to create a listing and pass its global name to Database.CreateFromListing instead of relying on env variable. We could go with a pre-created listing, but the preferred method is to create dependencies if possible.

**Review by sfc-gh-jcieslak — CHANGES_REQUESTED** [review]
Thank you for creating this PR, before merging, I would like you to address one thing. I think this should be achievable to create a listing in the test instead of relying on a pre-created one.

File: pkg/sdk/testint/databases_integration_test.go (hunk 1 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/databases_integration_test.go]: \+1

**Comment by ghribikhawla-prog** [comment]
Thanks for the feedback! I've updated the integration test to dynamically create a listing instead of relying on an env variable.

The test now:
1. Creates a Share and grants usage on a database
2. Creates a Listing with `WithTargetAccounts` pointing to the current account
3. Publishes the listing
4. Retrieves the `GlobalName` via `ShowByID`
5. Uses it with `CreateFromListing`
6. Cleans up everything via `t.Cleanup`

This follows the same pattern used in `listings_integration_test.go`.

File: pkg/sdk/testint/databases_integration_test.go [hunk 2]
```diff
@@ -337,6 +337,46 @@ func TestInt_DatabasesCreateSecondary(t *testing.T) {
...
+	share, shareCleanup := testClientHelper().Share.CreateShare(t)
+	t.Cleanup(shareCleanup)
+	t.Cleanup(testClientHelper().Grant.GrantPrivilegeOnDatabaseToShare(t, testClientHelper().Ids.DatabaseId(), share.ID(), []sdk.ObjectPrivilege{sdk.ObjectPrivilegeUsage}))
+
+	// Create a listing targeting the current account so we can consume it from the same account
+	accountId := testClientHelper().Context.CurrentAccountId(t)
+	manifest, _ := testClientHelper().Listing.BasicManifestWithTargetAccounts(t, accountId)
+
+	listingId := testClientHelper().Ids.RandomAccountObjectIdentifier()
+	err := client.Listings.Create(ctx, sdk.NewCreateListingRequest(listingId).
```
Comment (sfc-gh-jmichalak) [inline pkg/sdk/testint/databases_integration_test.go]: Nit: we could add a function like `testClientHelper().Listing.CreateWithTargetAccounts(...)`. which would use BasicManifestWithTargetAccounts as a manifest. The general principle is that for dependencies, we extract setups to `testClientHelper()`. Inside, they register cleanup functions and return the SDK objects.

**Review by sfc-gh-jmichalak — DISMISSED** [review]

**Comment by ghribikhawla-prog** [comment]
Good point, thanks! I'll extract that into a helper in the follow-up PR.

**Review by sfc-gh-jcieslak — APPROVED** [review]
Hey @ghribikhawla-prog 👋 
I had to adjust the integration tests as they seemed to be failing. Please, try to test the changes or just communicate that you are not able to run them (the test changes seem correct, but it turned out more setup was needed), and you can also run `make pre-push` (after setting it up with `make dev-setup`) to run certain validations and formatting as any differences with pre-push are detected by our CI. Other than that, the PR was solid, and thank you for your time 🙏. If you are planning to introduce the resource for this, I propose it be a new resource, as with object variants (e.g., look at function resources), we split the resources to avoid god resources with many different responsibilities. Other than that, the following contribution guide should help you with going forward.

**Review by sfc-gh-jmichalak — APPROVED** [review]
